### PR TITLE
BUG & FEAT Fix convergence issues with Pinball loss on large datasets (Issue #276)

### PR DESCRIPTION
**Pull request by floriankozikowski**
## Context of the PR
As reported in Issue #276, the PDCD_WS solver exhibits convergence problems with the Pinball loss on larger datasets, appearing to reach a saddle point state. The issue can be reproduced using:

```
import numpy as np
from skglm import GeneralizedLinearEstimator
from skglm.experimental.pdcd_ws import PDCD_WS
from skglm.experimental.quantile_regression import Pinball
from skglm.penalties import L1
from sklearn.datasets import make_regression
from sklearn.preprocessing import StandardScaler

# Generate data with 1000 samples (issue happens at n > 1000)
X, y = make_regression(n_samples=1000, n_features=10, noise=0.1, random_state=42)
X = StandardScaler().fit_transform(X)

# Set up problem with Pinball loss and PDCD_WS solver
datafit = Pinball(0.5)  # Median regression
penalty = L1(alpha=0.1)
solver = PDCD_WS(max_iter=500, max_epochs=500, tol=1e-2, verbose=True)

# This fit will not converge well
estimator = GeneralizedLinearEstimator(datafit=datafit, penalty=penalty, solver=solver)
estimator.fit(X, y)
```

## Contributions of the PR

1. Created a new QuantileHuber class - a smoothed version of the Pinball loss that replaces the non-differentiable point with a quadratic region, similar to how Huber loss smooths the absolute loss.
2. Implemented SmoothQuantileRegressor - a progressive smoothing solver that:
- Gradually decreases the smoothing parameter (delta)
- Solves each smoothed problem with FISTA
- Tracks the best solution (lowest quantile error) across all smoothing levels

Users can now solve quantile regression problems on larger datasets with:

```
from skglm.experimental.smooth_quantile_regressor import SmoothQuantileRegressor
import numpy as np
from sklearn.preprocessing import StandardScaler
from sklearn.datasets import make_regression

# Generate data
X, y = make_regression(n_samples=1000, n_features=10, noise=0.1)
X = StandardScaler().fit_transform(X)

# Create and fit regressor
sqr = SmoothQuantileRegressor(
    quantile=0.5,  # for median regression (0.5) or any other quantile
    alpha=0.1,     # L1 regularization strength
    verbose=False
)
sqr.fit(X, y)

# Make predictions
y_pred = sqr.predict(X)
```

## Limitations and Need for Further Refinement:
Currently, this is an approximation approach and NOT an exact solution. 
Speed has not been tested, and the unit test for now only successfully tests against Issue #276. 
Initial idea was to use a dual solver approach, where PDCD_WS its the Final Solver. This did not work.

Potential solutions could be path following methods, or review the code again (from earlier commits) and look for any mistakes (e.g. warmstart, intercept mistakes, etc.)

### Checks before merging PR

- [ ] added documentation for any new feature
- [ ] added unit tests
- [ ] edited the [what's new](../doc/changes/whats_new.rst) (if applicable)

### CODE REVIEW

File: examples/plot_smooth_quantile.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+"""
```
Comment (floriankozikowski) [inline examples/plot_smooth_quantile.py:1]: Some tests so far: 
- τ ≠ 0.5 (e.g. 0.8): SmoothQuantileRegressor reduces loss by >50% vs QuantileRegressor.  
- Large n (≥10 000):SmoothQuantileRegressor is 1.3×–2× faster *and* more accurate.  
- Median τ=0.5 & n≈1 000: scikit-learn’s QuantileRegressor remains the best choice.

These are anecdotal results—your mileage may vary. Tune sequence and inner‐solver settings accordingly. Still room for improvement

File: skglm/experimental/smooth_quantile_regressor.py [hunk 2]
```diff
@@ -0,0 +1,553 @@
+import numpy as np
+from scipy import sparse
+from skglm.experimental.pdcd_ws import PDCD_WS
+from numba import jit
+from skglm import GeneralizedLinearEstimator
+from skglm.penalties import L1
+from skglm.experimental.solver_strategies import StageBasedSolverStrategy
+
+
+@jit(nopython=True)
```
Comment (mathurinm) [inline skglm/experimental/smooth_quantile_regressor.py:10]: a priori there is no for loop, everything is vectorized so no need to JIT compile it

File: skglm/experimental/smooth_quantile_regressor.py [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+    elif quantile_error < 0.02 and gap < 0.1:
+        # Getting closer, reduce moderately
+        return max(current_delta * 0.85, min_delta)
+    else:
+        # Far from solution, reduce very conservatively
+        return max(current_delta * 0.9, min_delta)
+
+
+@jit(nopython=True, cache=True)
+def max_subgrad_gap(residuals, delta, quantile):
```
Comment (mathurinm) [inline skglm/experimental/smooth_quantile_regressor.py:39]: what is the purpose of this compared to existing skglm code ?

File: skglm/experimental/smooth_quantile_regressor.py [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+        max_stages=8,
+        quantile_error_threshold=0.005,
+        solver_params=None,
+        initial_alpha: float = None,
+        alpha_schedule: str = 'geometric',
+    ):
+        self.quantile = float(quantile)
+        if not 0 < self.quantile < 1:
+            raise ValueError("quantile must be between 0 and 1")
+        self.alpha = float(alpha)
```
Comment (mathurinm) [inline skglm/experimental/smooth_quantile_regressor.py:228]: look at other parts of the code,we do not cast stuff as float (a priori it's not needed, python will do it itself if need be)

File: skglm/experimental/smooth_quantile_regressor.py [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+            initial_alpha) if initial_alpha is not None else 5.0 * self.alpha
+        self.alpha_schedule = alpha_schedule
+
+        self.solver_params = {} if solver_params is None else dict(solver_params)
+
+        # Initialize solver strategy
+        self.solver_strategy = StageBasedSolverStrategy(self.solver_params)
+
+        from skglm.experimental.quantile_huber import QuantileHuber
+        self._quantile_huber_cls = QuantileHuber
```
Comment (mathurinm) [inline skglm/experimental/smooth_quantile_regressor.py:248]: what's the point of this if it has no attribute ?

File: skglm/experimental/smooth_quantile_regressor.py [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+                p0=200
+            )
+        else:
+            self.smooth_solver = smooth_solver
+
+        # Ensure consistent solver settings
+        if not hasattr(self.smooth_solver, 'warm_start'):
+            self.smooth_solver.warm_start = False
+
+    def _initialize_pdcd(self, X, y):
```
Comment (mathurinm) [inline skglm/experimental/smooth_quantile_regressor.py:266]: for a v0, use only smooth problems, so use AndersonCDWS solver instead. It should not have any params

File: skglm/experimental/smooth_quantile_regressor.py [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+        """Fit the model according to the given training data."""
+        # Basic validation
+        n_samples, n_features = X.shape
+        if len(y) != n_samples:
+            raise ValueError(f"X has {n_samples} samples, but y has {len(y)} samples")
+        is_sparse = sparse.issparse(X)
+
+        # Center data to handle intercept manually
+        y_mean = np.mean(y)
+        if is_sparse:
```
Comment (mathurinm) [inline skglm/experimental/smooth_quantile_regressor.py:325]: this is to be avoided at all cost ; all our solvers support sparse X ! also, no need to center
